Templates/Swift/Sources/APIClient.swift

```diff
@@ -39,11 +39,21 @@ public class APIClient {
     /// - Parameters:
     ///   - request: The API request to make
     ///   - behaviours: A list of behaviours that will be run for this request. Merged with APIClient.behaviours
+    ///   - dataPreprocessor:    `DataPreprocessor` which processes the received `Data` before calling the
+    ///                          `completionHandler`. `PassthroughPreprocessor()` by default.
+    ///   - emptyResponseCodes:  HTTP status codes for which empty responses are always valid. `[204, 205]` by default.
+    ///   - emptyRequestMethods: `HTTPMethod`s for which empty responses are always valid. `[.head]` by default.
     ///   - completionQueue: The queue that complete will be called on
     ///   - complete: A closure that gets passed the APIResponse
     /// - Returns: A cancellable request. Not that cancellation will only work after any validation RequestBehaviours have run
     @discardableResult
-    public func makeRequest<T>(_ request: APIRequest<T>, behaviours: [RequestBehaviour] = [], completionQueue: DispatchQueue = DispatchQueue.main, complete: @escaping (APIResponse<T>) -> Void) -> CancellableRequest? {
+    public func makeRequest<T>(_ request: APIRequest<T>,
+                               behaviours: [RequestBehaviour] = [],
+                               dataPreprocessor: DataPreprocessor = DataResponseSerializer.defaultDataPreprocessor,
+                               emptyResponseCodes: Set<Int> = DataResponseSerializer.defaultEmptyResponseCodes,
+                               emptyRequestMethods: Set<HTTPMethod> = DataResponseSerializer.defaultEmptyRequestMethods,
```

Comment:
Replicates the call site of Alamofire's request(...) method. I'm happy about any suggestions on how this could be improved.